src/controllers/wordlist-controller.js

```diff
@@ -1,122 +1,223 @@
+    events.DEFS_READY.sub(this.onDefinitionsReady.bind(this))
+    events.LEMMA_TRANSL_READY.sub(this.onLemmaTranslationsReady.bind(this))
   }
 
-  get availableLangs () {
-    return ['lat', 'grc', 'ara', 'per', 'gez']
+  /**
+   * Asynchronously initialize the word lists managed by this controller
+   * @param {UserDataManager} dataManager a user data manager to retrieve initial wordlist data from
+   *  // TODO may need a way to process a queue of pending words here e.g. if the wordlist controller isn't
+    * // activated until after number of lookups have already occurred
+   * Emits a WORDLIST_UPDATED event when the wordlists are available
+   */
+  async initLists (dataManager) {
+    for (let languageCode of this.availableLangs) {
+      let wordItems = await dataManager.query({dataType: 'WordItem', params: {languageCode: languageCode}})
+      if (wordItems.length > 0) {
+        this.wordLists[languageCode] = new WordList(languageCode,wordItems)
+      }
+    }
+    WordlistController.evt.WORDLIST_UPDATED.pub(this.wordLists)
   }
 
   /**
-   * This method executes in UIControler init method (at the end)
-   * It checks if storageAdapter is avalable (in our case it checks if IndexededDB works in the current browser)
-   * And if it is available, it executes openDatabase and 
-   * passes initDBStructure method for the case, when database doesn't exist (onupgradeneeded event)
-   * and uploadListsFromDB for the case when database already exists
+   * Get the wordlist for a specific language code
+   * @param {String} languageCode the language for the list
+   * @param {Boolean} create set to true to create the list of it doesn't exist
+   * Emits a WORDLIST_CREATED event if a new list is created
+   * @return {WordList} the wordlist
    */
-  async initLists () {
-    if (this.storageAdapter.available) {
-      await this.uploadListsFromDB()
+  getWordList (languageCode, create=true) {
+    if (create && ! this._wordListExist(languageCode)) {
+      let wordList = new WordList(languageCode,[])
+      this.wordLists[languageCode] = wordList
+      WordlistController.evt.WORDLIST_CREATED.pub(wordList)
     }
+    return this.wordLists[languageCode]
   }
 
-
-   /**
-   * This method loads data from the table in onsuccess event
-   * event is an argument, and event.target.result - is a database varaiable
-   * We are checking all available lists name (using userIDLangCode property, that defines wordlist)
-   * with the help of keyRange property condition - {indexName: 'userIDLangCode', value: listID, type: 'only' }
-   * and if it retrieves data successfully, it executes parseResultToWordList
+  /**
+   * Remove a wordlist for a specific language code and all if its items
+   * @param {String} languageCode the language for the list
+   * Emits a WORDLIST_DELETED event
    */
+  removeWordList (languageCode) {
+    let toDelete = this.wordLists[languageCode]
+    delete this.wordLists[languageCode]
```

Comment:
Maybe "safer" was not the best word to say it. To me, `Map` is a specialized object to store data. Plain objects can be used for both storing data (that's was the only way to do it pre-ES2015) and as a general purpose business object. The use of `Map` can improve readability of code (on my opinion) as it is an indication that this is something for storing data only. The safety I was talking about is that `map.delete(key)` would not allow to remove props that are not intended for data storage by accident.

Definitely, there is no clear cut solution when to use `Map` and when Object, But if this data is reactive, then I agree that Object is a better choice. I just did not realize about its reactivity (that's why I asked 🙂).

Reply:
I guess this could change though when we move to the Vuex structure for the wordlists, right? Because we will have a wordlist data module which is responsible for making the data available to the vue components, and that way we can keep design decisions separate -- i.e. use a map inside wordlist if that make sense for it, and an object in the Vuex store?

Reply:
Yes, I think that will be the right approach. For performance, we should expose only those fields to Vue reactivity areas (data & props) that are displayed by the UI at the moment, but no more than that. We should keep reactive props to the absolute minimum. 

So if we (as I assume) would display a list of only one language at a time, we probably should keep lists for all languages inside a Map object that is outside of reactivity areas: it could be custom props of a Vue object (accessible via `$options`) or a prop of a regular JS object. When language is changed, we should assign a list for a selected language to the Vuex store module thus exposing it to UI components.

Would that make sense? What do you think about this approach?